frontend/src/vault/qr/upload/useProcessQrMutation.ts

```diff
@@ -0,0 +1,109 @@
+import { useMutation } from '@tanstack/react-query';
+import jsQR from 'jsqr';
+import { getRawQueryParams } from '../../../lib/utils/query/getRawQueryParams';
+import { decompressQrPayload } from './utils/decompressQrPayload';
+import { match } from '../../../lib/utils/match';
+import { Vault } from '../../../gen/vultisig/vault/v1/vault_pb';
+import { SaveVault } from '../../../../wailsjs/go/storage/Store';
+import { useNavigate } from 'react-router-dom';
+
+type QrTtsType = 'Keygen' | 'Reshare';
+
+type QrType = 'NewVault' | 'Keysign';
+
+type QrSharedData = {
+  jsonData: string;
+  vault: string;
+};
+
+type QrQueryParams =
+  | (QrSharedData & {
+      type: QrType;
+    })
+  | (QrSharedData & {
+      ttsType: QrTtsType;
+    });
+
+function stringToUint8Array(input: string): Uint8Array {
+  const encoder = new TextEncoder();
+  return encoder.encode(input);
+}
+
+export const useProcessQrMutation = () => {
+  const navigate = useNavigate();
+  return useMutation({
+    mutationFn: async (file: File) => {
+      const imageBitmap = await createImageBitmap(file);
+
+      // Create a canvas to draw the image
+      const canvas = document.createElement('canvas');
+      canvas.width = imageBitmap.width;
+      canvas.height = imageBitmap.height;
+
+      const context = canvas.getContext('2d');
+      if (!context) {
+        throw new Error('Could not get canvas context');
+      }
+
+      context.drawImage(imageBitmap, 0, 0);
+
+      const imageData = context.getImageData(0, 0, canvas.width, canvas.height);
+
+      const code = jsQR(imageData.data, imageData.width, imageData.height);
+
+      if (!code) {
+        throw new Error('Failed to read QR code');
+      }
+
+      const url = code.data;
+
+      const queryParams = getRawQueryParams<QrQueryParams>(url);
+      const { jsonData } = queryParams;
+
+      const payload = await decompressQrPayload(jsonData);
+      console.log(payload);
+
+      if ('type' in queryParams) {
+        return match(queryParams.type, {
+          NewVault: async () => {
+            const vault = Vault.fromBinary(stringToUint8Array(payload));
```

Comment:
can you remove this , the logic is wrong , because the payload is not a vault , it is `KeygenMessage` or `KeysignMessage`